ledger/src/blockstore_meta.rs

```diff
@@ -131,6 +135,39 @@ pub struct ErasureMeta {
     config: ErasureConfig,
 }
 
+// Helper module to serde values by type-casting to an intermediate
+// type for backward compatibility.
+mod serde_compat_cast {
+    use super::*;
+
+    // Serializes a value of type T by first type-casting to type R.
+    pub(super) fn serialize<S: Serializer, R, T: Copy>(
+        &val: &T,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error>
+    where
+        R: TryFrom<T> + Serialize,
+        <R as TryFrom<T>>::Error: std::fmt::Display,
+    {
+        R::try_from(val)
+            .map_err(serde::ser::Error::custom)?
+            .serialize(serializer)
+    }
+
+    // Deserializes a value of type R and type-casts it to type T.
+    pub(super) fn deserialize<'de, D, R, T>(deserializer: D) -> Result<T, D::Error>
+    where
+        D: Deserializer<'de>,
+        R: Deserialize<'de>,
+        T: TryFrom<R>,
+        <T as TryFrom<R>>::Error: std::fmt::Display,
+    {
+        R::deserialize(deserializer)
+            .map(T::try_from)?
+            .map_err(serde::de::Error::custom)
+    }
```

Comment:
intentionally generic so that it can be used elsewhere for similar type discrepancies.